pkg/jobstore/boltdb/store.go

```diff
@@ -290,19 +291,28 @@ func (b *BoltJobStore) jobExists(tx *bolt.Tx, jobID string) bool {
 }
 
 // GetJobs returns all Jobs that match the provided query
-func (b *BoltJobStore) GetJobs(ctx context.Context, query jobstore.JobQuery) ([]models.Job, error) {
-	var jobs []models.Job
+func (b *BoltJobStore) GetJobs(ctx context.Context, query jobstore.JobQuery) (*jobstore.JobQueryResponse, error) {
+	var response *jobstore.JobQueryResponse
 	err := b.database.View(func(tx *bolt.Tx) (err error) {
-		jobs, err = b.getJobs(tx, query)
+		response, err = b.getJobs(tx, query)
 		return
 	})
-	return jobs, err
+	return response, err
 }
 
-func (b *BoltJobStore) getJobs(tx *bolt.Tx, query jobstore.JobQuery) ([]models.Job, error) {
+func (b *BoltJobStore) getJobs(tx *bolt.Tx, query jobstore.JobQuery) (*jobstore.JobQueryResponse, error) {
 	if query.ID != "" {
 		job, err := b.getJob(tx, query.ID)
-		return []models.Job{job}, err
+
+		if query.Selector != nil {
+			// If we have a selector and it doesn't match, then it isn't an error
+			// just a non-match.
+			if !query.Selector.Matches(labels.Set(job.Labels)) {
+				return &jobstore.JobQueryResponse{}, nil
+			}
+		}
+
+		return &jobstore.JobQueryResponse{Jobs: []models.Job{job}}, err
```

Comment:
I think for historical reasons we used the same implementation for `get` and `list`, where `query.ID` was what differentiated between both. I don't think it makes sense to list jobs using `query.ID` as the ID is supposed to be unique. 

We should either remove `query.ID`, or at least don't add more features to it by allowing `query.ID` and `query.Selector` to mean something. What do you think?

Reply:
I think we should remove `query.ID` and just fix up the legacy usage of it until we remove it.  I think removing IncludeTags and ExcludeTags would also be good, but might be hard to replicate unless there's a clear mapping from include/exclude to a selector.

Reply:
Fixed in 30db3a3f0